### PR TITLE
Update publish docs in README

### PR DESCRIPTION
**Pull request by shilman**
Improved the step-by-step explanation of `lerna publish`.

## Description

1. Ordered list indentation does not work properly on Github, so even though the markdown is correct, it renders wrong making the output super confusing. I tried multiple permutations to try to get it to indent properly, and this was the one that worked (though the original markdown is better).
2. As I understand it, it is adding/removing package versions to the dist-tag, not tags.

## Motivation and Context

The steps are misleading in their current form.

## How Has This Been Tested?

N/A

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,14 +250,14 @@ Creates a new git commit/tag in the process of publishing to npm.
 
 More specifically, this command will:
 
-1. Publish each module in `packages` that has been updated since the last version to npm with the [dist-tag](https://docs.npmjs.com/cli/dist-tag) `lerna-temp`.
+- First, publish each module in `packages` that has been updated since the last version to npm with the [dist-tag](https://docs.npmjs.com/cli/dist-tag) `lerna-temp`.
```
Comment (evocateur) [inline README.md]: This temporary dist-tag strategy was actually inverted recently, which means that by default we _do not_ publish to the `lerna-temp` dist-tag first. Looks like we forgot a part of the docs to update!

I think we could make this much easier to read by entirely removing the outer list, so all that remains is the inner numbered list, which you should probably un-indent so Github doesn't try to get too fancy with it.

File: README.md [hunk 2]
```diff
@@ -250,14 +250,14 @@ Creates a new git commit/tag in the process of publishing to npm.
...
   1. Run the equivalent of `lerna updated` to determine which packages need to be published.
   2. If necessary, increment the `version` key in `lerna.json`.
   3. Update the `package.json` of all updated packages to their new versions.
   4. Update all dependencies of the updated packages with the new versions, specified with a [caret (^)](https://docs.npmjs.com/files/package.json#dependencies).
   5. Create a new git commit and tag for the new version.
   6. Publish updated packages to npm.
-2. Once all packages have been published, remove the `lerna-temp` tags and add the tags to `latest`.
+- Second, after all packages have been published, remove the `lerna-temp` versions and add the versions to `latest`.
 
 > A temporary dist-tag is used at the start to prevent the case where only some of the packages are published; this can cause issues for users installing a package that only has some updated packages.
```
Comment (evocateur) [inline README.md]: Oh, and remove this temporary dist-tag blurb, while you're at it.

**Comment by shilman** [comment]
@evocateur updated per your comments! thanks!

**Comment by evocateur** [comment]
@shilman Thanks for your patience!

**Pull request merged**